### PR TITLE
fix: correct Utf8String encoding and dynamic array handling for non-ASCII strings (#1841)

### LINKED ISSUE #1841: utf8string[] can not process the chinese words correctly
When I set the metadata for a erc721 contract, the `name` and the `description` are set to the CHINESE words. If the string length is too long, the tx will be reverted. Although the string length is short, for example, 2 chinese words, the corresponding field in the metadata  will be shown incorrectly in the scan. How to fix it?

will be reverted one:
image

the incorrect one:
image

### PR DESCRIPTION
**Pull request by Dev10-sys**
Fixes incorrect handling of UTF-8 strings (e.g., Chinese text) which could lead to wrong encoding or transaction failures. 

The string length calculation has been updated to use actual UTF-8 byte length with proper ABI padding and array encoding has been corrected to handle static arrays containing dynamic types properly. 

Additional tests have been added to cover non-ASCII strings and edge cases.

Fixes #1841

### CODE REVIEW

File: abi/src/test/java/org/web3j/abi/datatypes/ChineseUtf8StringTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package org.web3j.abi.datatypes;
```
Comment (NickSneo) [inline abi/src/test/java/org/web3j/abi/datatypes/ChineseUtf8StringTest.java:13]: Missing web3j license headers

File: abi/src/test/java/org/web3j/abi/Utf8StringTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
+package org.web3j.abi;
```
Comment (NickSneo) [inline abi/src/test/java/org/web3j/abi/Utf8StringTest.java:13]: Same, missing web3j license headers

File: abi/src/test/java/org/web3j/abi/datatypes/ChineseUtf8StringTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+public class ChineseUtf8StringTest {
+
+    @Test
+    public void testChineseStringEncoding() {
+        // Chinese text: Zhong Guo Liu Li Fa Lang Cai Hua Niao Zi Ming Zhong
+        String chineseText = "\u4e2d\u56fd\u7409\u7483\u73d0\u7405\u5f69\u82b1\u9e1f\u81ea\u9e23\u949f";
+        Utf8String utf8String = new Utf8String(chineseText);
+        
+        String encoded = TypeEncoder.encode(utf8String);
+        System.out.println("Encoded: " + encoded);
```
Comment (NickSneo) [inline abi/src/test/java/org/web3j/abi/datatypes/ChineseUtf8StringTest.java]: Console logs should be removed, instead add assert tests

File: abi/src/test/java/org/web3j/abi/datatypes/ChineseUtf8StringTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        
+        byte[] bytes = chineseText.getBytes(StandardCharsets.UTF_8);
+        System.out.println("Byte length: " + bytes.length);
```
Comment (NickSneo) [inline abi/src/test/java/org/web3j/abi/datatypes/ChineseUtf8StringTest.java]: same, remove `System.out.println`

File: abi/src/test/java/org/web3j/abi/datatypes/ChineseUtf8StringTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Test
+    public void testChineseStringInArray() {
+        // Chinese text: Zhong Guo
+        String chineseText = "\u4e2d\u56fd";
+        Utf8String utf8String = new Utf8String(chineseText);
+        DynamicArray<Utf8String> array = new DynamicArray<>(Utf8String.class, utf8String);
+        
+        String encoded = TypeEncoder.encode(array);
+        System.out.println("Array Encoded: " + encoded);
```
Comment (NickSneo) [inline abi/src/test/java/org/web3j/abi/datatypes/ChineseUtf8StringTest.java]: same, remove System.out.println

**Comment by Dev10-sys** [comment]
@NickSneo, I have made all the suggested changes.

**Review by NickSneo — APPROVED** [review]
LGTM
